ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5008,11 +5009,25 @@ static private boolean needToCopy(final HiveConf conf, Path srcf, Path destf, Fi
         boolean isOwned = FileUtils.isOwnerOfFileHierarchy(srcFs, srcs, configuredOwner, false);
         if (configuredOwner.equals(runningUser)) {
           // Check if owner has write permission, else it will have to copy
+          UserGroupInformation ugi = Utils.getUGI();
+          String currentUser = ugi.getShortUserName();
+          FileSystem fsAsUser = null;
+          UserGroupInformation proxyUser = null;
+          if (configuredOwner != null && !configuredOwner.equals(currentUser)) {
+            proxyUser = UserGroupInformation.createProxyUser(configuredOwner, UserGroupInformation.getLoginUser());
```

Comment:
This block is used more than once. Can be moved to a common block.

Reply:
Was thinking about this. But proxyUser is needed for the finally block to close FS connections.

Reply:
I think this can be done in two ways ..
1. Use a common function to get proxy user and then in getFsAsUser use null check for proxy user
2. Use a Pair to return

Reply:
I used the option 1 :)